src/java/org/apache/cassandra/io/util/CompressedChunkReader.java

```diff
@@ -83,20 +89,38 @@ public BufferType preferredBufferType()
     }
 
     @Override
-    public Rebufferer instantiateRebufferer()
+    public Rebufferer instantiateRebufferer(boolean isScan)
     {
-        return new BufferManagingRebufferer.Aligned(this);
+        return new BufferManagingRebufferer.Aligned(this.setMode(isScan));
     }
 
     public static class Standard extends CompressedChunkReader
     {
         // we read the raw compressed bytes into this buffer, then uncompressed them into the provided one.
         private final ThreadLocalByteBufferHolder bufferHolder;
+        private final ThreadLocalReadAheadBuffer readAheadBuffer;
+
 
         public Standard(ChannelProxy channel, CompressionMetadata metadata, Supplier<Double> crcCheckChanceSupplier)
         {
             super(channel, metadata, crcCheckChanceSupplier);
             bufferHolder = new ThreadLocalByteBufferHolder(metadata.compressor().preferredBufferType());
+
+            int readAheadBufferSize = DatabaseDescriptor.getCompressedReadAheadBufferSize();
+            readAheadBuffer = readAheadBufferSize > 0 ? new ThreadLocalReadAheadBuffer(channel, readAheadBufferSize, metadata.compressor().preferredBufferType()) : null;
+        }
+
+        @Override
+        public CompressedChunkReader setMode(boolean isScan)
+        {
+            if (readAheadBuffer != null)
```

Comment:
nit: Quick comment on the fact that `isScan` is there so we don't try to use the RAB on point reads